### DISCUSSION TITLE
JSON: Allow separators at the ends of arrays, objects

### DISCUSSION THREAD

**Discussion by actual-nh** [CleverRaven/Cataclysm-DDA#46339] [Ideas]
Currently, the JSON validator/formatter errors at having a separator (",") at the end of any array or object. While strictly correct according to the format, this makes it much easier to forget **_adding_** a separator when adding new objects (or array lines). I suggest allowing "extra" separators at these places, and possibly even having the autoformatter automatically add them (at minimum, it should not remove them; simply adding them in any event is likely the easiest way to do this).

**Comment by anothersimulacrum** [comment]
I don't think deviating from the JSON standard is a very good idea. If we're not compliant with it, it will make it harder or even impossible to use external tools that deal with JSON.

**Reply by actual-nh** [reply]
I note the error messages are saying "strictly" - does this indicate that there are different levels of compliance (as with loose vs strict in various versions of HTML)? (Just hoping... sigh.)

**Reply by anothersimulacrum** [reply]
I don't know of anything regarding that, but I am not super familiar with the JSON standard/use of it.
I know that [jsonlint.com](https://jsonlint.com/) will not accept that, and that's a pretty reasonable tool to use to find syntax errors in JSON (because our linter on the website doesn't do it (though the command line version does)).

**Comment by Aivean** [comment]
At least some IDEs (e.g. Clion) would report such json as incorrect:
![image](https://user-images.githubusercontent.com/2865203/103159921-1811b180-4784-11eb-9e23-9ffe3020e28e.png)

**Comment by actual-nh** [comment]
Sigh. OK. Unfortunate.
